### PR TITLE
[ADD] res_partner_operating_unit

### PR DESCRIPTION
**Pull request by repodevs**
This module introduces Operating Unit in res partner

### CODE REVIEW

**Comment by repodevs** [comment]
@aheficent Could you please review this? 
Runbot is ok, but I don't know why travis is failing

I'm quite new in all this, thanks for understanding. (:

**Review by AaronHForgeFlow — COMMENTED** [review]
Hi @repodevs, thank you for your module. Code seems ok to me.  Travis is failing because the changes introduced here make the other tests in this repo to fail. I'm not sure but maybe it's the default value for the operating unit assigned to the partner.

BTW, have you signed the CLA?

**Comment by repodevs** [comment]
@aheficent Yes, the default Operating Unit is assigned to the partner based on their default Operating Unit (in res.users settings). Should default Operating Unit is not assigned in partner?

Yes I have signed the CLA, you can check on [OCA web portal](https://odoo-community.org/web#id=3637&view_type=form&model=res.partner&action=773)

**Comment by repodevs** [comment]
@aheficent ahaa, I think the default Operating Unit is must not assigned to the partner. Other tests module is used partner that not have operating unit assigned. so that's will make the tests fail

**Comment by repodevs** [comment]
@aheficent runbot and travis now is ok :)

File: res_partner_operating_unit/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+{
+    "name": "Partner with Operating Unit",
+    "summary": "Introduces Operating Unit fields in Partner",
+    "version": "10.0.1.0.0",
+    "author": "Edi Santoso, "
+              "Niaga Solution, "
+              "Odoo Community Association (OCA)",
+    "website": "https://www.niagasolution.com",
+    "category": "Generic",
+    "depends": ["base"],
```
Comment (AaronHForgeFlow) [inline res_partner_operating_unit/__manifest__.py]: It should depend on operating_unit module

**Review by AaronHForgeFlow — COMMENTED** [review]
I tested on runbot and It works fine. Can you add a test to check users don't access partners they are not allowed to? I'll approve after that :smile:

**Comment by repodevs** [comment]
@aheficent manifest fixed and tests added :smile:

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: Thank you for attending the comments

**Review by AaronHForgeFlow — CHANGES_REQUESTED** [review]
Hi @repodevs, I'm afraid I have to dismiss my last review.

From a functional perspective, this module is limiting the number of operating units associated to a partner to one. This is not correct because a partner may be related to several operating units. Two operating units may share customers but a third operating unit may not use that customer.

**Comment by repodevs** [comment]
@aheficent should we change to `many2many` instead of `many2one` ?

**Comment by AaronHForgeFlow** [comment]
@repodevs I think so

File: res_partner_operating_unit/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+# © 2017 Niaga Solution - Edi Santoso <repodevs@gmail.com>
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+{
+    "name": "Partner with Operating Unit",
+    "summary": "Introduces Operating Unit fields in Partner",
+    "version": "10.0.1.0.0",
+    "author": "Edi Santoso, "
+              "Niaga Solution, "
+              "Odoo Community Association (OCA)",
+    "website": "https://www.niagasolution.com",
```
Comment (nikul-serpentcs) [inline res_partner_operating_unit/__manifest__.py:11]:
```suggestion
    "website": "https://github.com/OCA/operating-unit",
```

File: res_partner_operating_unit/models/res_partner.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+from odoo import fields, models
+
+
+class ResPartner(models.Model):
+
+    _inherit = 'res.partner'
+
+    operating_unit_id = fields.Many2one(
+        comodel_name='operating.unit', string='Operating Unit'
+    )
```
Comment (max3903) [inline res_partner_operating_unit/models/res_partner.py:13]: What about a many2many?

**Pull request closed without merging**